doxygen/contributor_help_pages/require_meta.md

```diff
@@ -0,0 +1,191 @@
+
+```cpp
+ // Works if either neither T1 or T2 are arithmetic
+ template <typename T1, typename T2,
+   require_not_any_eigen_row_vector_t<T1, T2>* = nullptr>
+ auto add(T1 x, T2 y) { return x + y; }
+```
+
+- `require_not_all_*_t` : Takes a parameter pack of types to enable if all
+of the types are not satisfied.
+
+```cpp
+ // Works if neither T1 and T2 are arithmetic
+ template <typename T1, typename T2,
+   require_not_all_arithmetic_t<T1, T2>* = nullptr>
+ auto add(T1 x, T2 y) { return x + y; }
+```
+
+ `std::vector` and `Eigen` types have additional requires to
+detect if the @ref stan::value_type (the first underlying type) or the
+ @ref stan::scalar_type
+(the containers underlying scalar type) satisfy a condition to enable a class
+or function.
+
+ The container `requires` have an ending at their signature of _vt and _st to symbolize whether you want to inspect the @ref stan::value_type or @ref stan::scalar_type. A function
+that accepts eigen matrices with floating point value types can be defined as
+
+```cpp
+ template <typename Mat1, typename Mat2,
+   require_all_eigen_vt<std::is_floating_point, Mat1, Mat2>* = nullptr>
+ auto add(Mat1&& A, Mat2&& B) { return A + B;}
+```
+  A function that accepts standard vectors of Eigen vectors whose scalar type
+is @ref stan::math::var types can be defined as
+
+```cpp
+ template <typename Vec1, typename Vec2,
+   require_all_std_vector_vt<is_eigen_vector, Vec1, Vec2>* = nullptr,
+   require_all_std_vector_st<is_var, Vec1, Vec2>* = nullptr>
+ auto add(Vec1&& A, Vec2&& B) {
+   std::vector<decltype<A[0] + B[0]>> return_vec;
+   std::transform(A.begin(), A.end(), B.begin(), return_vec.begin(),
+     [](auto&& x, auto&& y) {
+         return x + y;
+     });
+   return return_vec;
+ }
+```
+
+ There are also requires for generically checking if a type's @ref stan::value_type or
+  @ref stan::scalar_type is correct. To differentiate them from the Eigen and Standard
+ vector checks the `vt` and `st` comes *before* the type such as
+ `require_vt_var<T>` which checks if a type `T`'s @ref stan::value_type satisfies
+  @ref stan::is_var.
+
+ The `requires` type traits allow Stan to have more generic types so that the
+ library can forward along Eigen expression and have better move semantics.
+ For instance, the code below will accept any arbitrary Eigen expression
+ that, if it's an rvalue, can be forwarded to another function.
+
+```
+ template <typename Mat1, typename Mat2,
+  require_all_eigen_vt<is_arithmetic, Mat1, Mat2>* = nullptr>
+ inline auto a_func(Mat1&& m1, Mat2&& m2) {
+   check_nan(m1);
+   check_nan(m2);
+   return another_func(std::forward<Mat1>(m1), std::forward<Mat2>(m2));
+```
+
+#### Adding a new requires
+
+For a full list of predefined `requires` please see @ref require_meta.
+
+If you are adding a new type trait that returns has a `bool value` member you can add it to the set of known requires by using the macros @ref stan/math/prim/meta/require_helpers.hpp .
```

Comment:
returns has -> returns

`bool value` member -> `bool value` (members don't get returned by functions)

macros -> macros defined in

Reply:
Fixed. This should not have been about returns but instead about definint type traits that have a `bool` member named `value`.